### PR TITLE
Exposed YARN node address in the ProtoBuf and Python API

### LINKED ISSUE #38: Expose node metadata in ``Container``
Currently the `skein.Container` class only contains the container ID. It might be handy to expose other metadata [available](https://hadoop.apache.org/docs/r2.6.1/api/org/apache/hadoop/yarn/api/records/Container.html), e.g. node ID or node HTTP address.

This is needed for #34.

### PR DESCRIPTION
**Pull request by superbobry**
Closes #38

I thought about adding a `NodeId` message to the ProtoBuf, but a `str` is simpler and seems to be sufficient for now, wdyt?

### CODE REVIEW

File: skein/model.py [hunk 1]
```diff
@@ -1038,23 +1038,26 @@ class Container(Base):
...
     start_time : datetime
         The start time, None if container has not started.
     finish_time : datetime
         The finish time, None if container has not finished.
     """
     __slots__ = ('service_name', 'instance', '_state', 'yarn_container_id',
-                 'start_time', 'finish_time')
+                 'yarn_node_address', 'start_time', 'finish_time')
     _params = ('service_name', 'instance', 'state', 'yarn_container_id',
                'start_time', 'finish_time')
```
Comment (jcrist) [inline skein/model.py]: Need to also add it to `_params`

File: skein/test/test_model.py [hunk 2]
```diff
@@ -348,7 +348,8 @@ def test_container():
 
     kwargs = dict(service_name="foo",
                   instance=0,
-                  yarn_container_id='container_1528138529205_0038_01_000001')
+                  yarn_container_id='container_1528138529205_0038_01_000001'
+                  yarn_node_address='localhost:14420')
```
Comment (jcrist) [inline skein/test/test_model.py]: you're missing a comma above ^^, this is a syntax error.

File: skein/model.py (hunk 1 above)
Comment (superbobry) [reply skein/model.py]: How about removing `_params` here? `_params == __slots__` so `_get_params` would behave as expected.

File: skein/test/test_model.py (hunk 2 above)
Comment (superbobry) [reply skein/test/test_model.py]: Ouch, fixed.

File: skein/model.py (hunk 1 above)
Comment (jcrist) [reply skein/model.py]: > `_params == __slots__` so `_get_params`

There's a difference between `'_state'` vs `state'`.

**Comment by jcrist** [comment]
This looks good to me. Thanks!

**Pull request merged**